doc/md/fundamentals/1-basic-syntax/8-functions.md

```diff
@@ -5,40 +5,79 @@ hide_table_of_contents: true
 
 # Functions
 
-Functions in Motoko can take on a variety of different attributes, the most basic of which is if the function is public or private. Public functions are callable by users or other [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters), while private functions can only be called within the program that defines it.
+Functions in Motoko can have various attributes, the most fundamental being whether they are public or private. Public functions can be called by users or other [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters), while private functions are only accessible within the program that defines them.
 
 The most basic Motoko [function declaration](https://internetcomputer.org/docs/motoko/fundamentals/declarations/function-declarations) is:
 
-```motoko
-func exampleFunction() {};
+```motoko no-repl
+func exampleFunction() : () {};
 ```
 
-By default, all functions are declared private unless explicitly declared `public`.
+In objects, modules, and actors, all functions are private by default unless explicitly declared as `public`.
 
-```motoko no-repl
-public func exampleFunction() {};
+```motoko
+object Counter  {
+   var value = 0;
+   func reset() { value := 0 };
+   public func inc() { value := 1};
+   public func get() : Nat { value }; 
+}
 ```
 
-A function should specify a return type. If a return type is not declared, it defaults to the unit `()` return type.
+The object `Counter` has two public methods, the functions `Counter.inc()` and `Counter.get()`. Both `value` and `reset()` are implicitly `private`. Any attempts to access `Counter.reset()` and `Counter.value` produce type errors.
 
-```motoko no-repl
+A function should specify a return type. If a return type is not declared or otherwise determined from the context, it defaults to the unit `()` return type.
+
+```motoko
 public func exampleFunction(x : Nat) : Nat {
```

Comment:
```suggestion
func exampleFunction(x : Nat) : Nat {
```
(otherwise it doesn't work in the repl. Only object/module and actor members can have the keyword `public`.